runtime/vm/CRIUHelpers.cpp

```diff
@@ -547,15 +547,17 @@ getRestoreSystemProperites(J9VMThread *currentThread)
 
 	if (NULL != vm->checkpointState.restoreArgsList) {
 		JavaVMInitArgs *restorArgs = vm->checkpointState.restoreArgsList->actualVMArgs;
-		 J9CmdLineOption* j9Options = vm->checkpointState.restoreArgsList->j9Options;
-		J9MemoryManagerFunctions * mmfns = vm->memoryManagerFunctions;
+		 J9CmdLineOption *j9Options = vm->checkpointState.restoreArgsList->j9Options;
+		J9MemoryManagerFunctions *mmfns = vm->memoryManagerFunctions;
 		J9InternalVMFunctions *vmFuncs = vm->internalVMFunctions;
 		PORT_ACCESS_FROM_JAVAVM(vm);
 
 		UDATA count = 0;
+		IDATA index = 0;
```

Comment:
my preference is to have loop variables has part of the loop decl when possible

Reply:
`restorArgs->options` contains `-Xoptionsfile=optionsFilePath` at the first entry. `count` is calculated correctly, but `index` was incremented regardless of entry type, hence the first element is null, and caused the NPE
```
Exception in thread "main" java.lang.NullPointerException
	at java.base/java.lang.System.setProperty(System.java:633)
	at openj9.criu/org.eclipse.openj9.criu.CRIUSupport.setRestoreJavaProperties(CRIUSupport.java:600)
	at openj9.criu/org.eclipse.openj9.criu.J9InternalCheckpointHookAPI$J9InternalCheckpointHook.runHook(J9InternalCheckpointHookAPI.java:116)
	at openj9.criu/org.eclipse.openj9.criu.J9InternalCheckpointHookAPI.runHooks(J9InternalCheckpointHookAPI.java:80)
	at openj9.criu/org.eclipse.openj9.criu.J9InternalCheckpointHookAPI.runPostRestoreHooks(J9InternalCheckpointHookAPI.java:97)
	at openj9.criu/org.eclipse.openj9.criu.CRIUSupport.checkpointJVMImpl(Native Method)
	at openj9.criu/org.eclipse.openj9.criu.CRIUSupport.checkpointJVM(CRIUSupport.java:635)
	at org.openj9.criu.CRIUTestUtils.checkPointJVM(CRIUTestUtils.java:77)
	at org.openj9.criu.OptionsFileTest.propertiesTest1(OptionsFileTest.java:68)
	at org.openj9.criu.OptionsFileTest.main(OptionsFileTest.java:38)
```

This NPE is not explicitly caught as a failure condition, and the test passed.

`-Dprop1=val1\n-Dprop2=\\\nval2\n-Dprop3=v \\ \n a \\     \n  l3` within `propertiesTest3()` caused 
```
JVMJ9VM007E Command-line option unrecognised: a\
Exception in thread "main" org.eclipse.openj9.criu.JVMRestoreException: The JVM could not enable all the restore options specified
	at openj9.criu/org.eclipse.openj9.criu.CRIUSupport.checkpointJVMImpl(Native Method)
	at openj9.criu/org.eclipse.openj9.criu.CRIUSupport.checkpointJVM(CRIUSupport.java:635)
	at org.openj9.criu.CRIUTestUtils.checkPointJVM(CRIUTestUtils.java:77)
	at org.openj9.criu.OptionsFileTest.propertiesTest3(OptionsFileTest.java:121)
	at org.openj9.criu.OptionsFileTest.main(OptionsFileTest.java:44)
```

Reply:
I see, Index can still be declared in in the loop decl though

Reply:
`NullPointerException` should be a failure condition for all props test, this is what I meant by "it should be explicit"

Reply:
ah, I see youve dont that

Reply:
Moved the `Index` declaration into the loop.